### ISSUE TITLE
Updating to Fluent Assertions 6.0

### ISSUE THREAD

**Issue by jnick422** [serilog-contrib/SerilogSinksInMemory#22]
Recently updated a repository to Fluent Assertions 6.0 and was running into some issues with the InMemoryAssertions:
```
 Message: 
    System.MissingMethodException : Method not found: 'Void FluentAssertions.Primitives.ReferenceTypeAssertions`2..ctor()'.

  Stack Trace: 
    InMemorySinkAssertions.ctor(InMemorySink instance)
    InMemorySinkAssertionExtensions.Should(InMemorySink instance)
    CreateFromContractMessageHandlerTests.ShouldLogIfInvalidAccountName(Mock`1 s3Client) line 344
    --- End of stack trace from previous location ---
```

When making a call like so:

```csharp
InMemorySink.Instance.Should()
    .HaveErrorMessageWithException<ArgumentException>(
        "An invalid argument was encountered while processing message with ID {MessageId}",
        "Account name cannot be longer than 100 characters");
```

This is what the error message extension method looks like:
```csharp
      public static void HaveErrorMessageWithException<T>(this InMemorySinkAssertions assertion, string messageTemplate, string innerMessage = null)
      {
          if (innerMessage == null)
          {
              assertion.HaveMessage(messageTemplate)
                  .WithLevel(LogEventLevel.Error)
                  .Appearing().Once()
                  .Match(o => o.Exception.GetType() == typeof(T));
          }

          assertion.HaveMessage(messageTemplate)
              .WithLevel(LogEventLevel.Error)
              .Appearing().Once()
              .Match(o => o.Exception.GetType() == typeof(T) &&
                      o.Exception.Message.Contains(innerMessage)
              );
      }
```
This was working with Fluent Assertions 5.10.3. Let me know if you need any more information

**Comment by sandermvanvliet** [comment]
This is not uncommon when FluentAssertiosn has an upgrade (and especially with major version updates).
I can take a look at this and see if we can make it compatible between versions of FluentAssertions

**Comment by sandermvanvliet** [comment]
@siewers @jnick422 sorry it took a while to get some time to do this.
I've managed to make this work without having to break compatibility with FluentAssertions 5.

However I do need to sort out some things with the build after moving this repo into serilog-contrib so it might take some time before this version hits NuGet.

**Comment by sandermvanvliet** [comment]
@jnick422 @siewers I've fixed the build issues and version 0.9.0 of the assertions package has just been published to NuGet so you should see it appearing in the NuGet feed shortly.

Thanks for bringing this issue to my attention and happy testing :+1:
